scripts/extract-extension.js

```diff
@@ -43,7 +43,11 @@ exports.extractExtension = async function (
     // Write the extension to the community extensions folder
     await pipeline(
       file.nodeStream(),
-      createWriteStream(`${extensionsFolder}/community/${file.name}`)
+      createWriteStream(
+        `${extensionsFolder}/${reviewed ? 'reviewed' : 'community'}/${
+          file.name
+        }`
+      )
```

Comment:
What do you think about asking submitters if they want to submit a update or a new extension?
- a new extension would always target the community folder
- an update would target either reviewed or community according to the extension to update.

The PR title or tags could make it clear that it's a reviewed extension update so that reviewers don't merge something in the reviewed folder by mistake without a full review.
It will also avoid to move extensions to the community folder because submitters chose "reviewed" whereas it's not up to them to decide and 90% of submission are likely to go to the community folder at least as a 1st step.

Reply:
I'd love a system where we have an issue template to submit a new extension that won't allow using a name that's already in-use, andcreates a community extensions PR by default, with another issue template to later on request an upgrade to reviewed tier, and an update extension issue template that'll take into account the current tier of the extension you're trying to update.

That way, we encourage community extensions by default, provide an upgrade path to reviewed extensions, and can still submit updates to all types of extensions without risking overriding/"updating" another one by accident, or decreasing the quality of the reviewed extension demoting it to community extension.

---

With that said, this PR already has a lot of changes. I'd rather first merge this one for now, to already get these improvements out and figure out any potential bug with these changes, then work on reworking & improving the submission system even more.

Reply:
I agree with your target, but I don't want to see new extension PR adding their file in the review folder while you implement the 2nd step because it will require me to manually fix these PR.